### PR TITLE
gossip/simulation: unexport unnecessary exports

### PR DESCRIPTION
**Pull request by tamird**
Discovered using github.com/jgautheron/usedexports.

The following false positives remain:

```
$ usedexports ./...
gossip/simulation/network.go:105:1 func "StartNode" is exported but not used locally
gossip/simulation/network.go:155:1 func "SimulateNetwork" is exported but not used locally
gossip/simulation/network.go:185:1 func "Stop" is exported but not used locally
gossip/simulation/network.go:195:1 func "RunUntilFullyConnected" is exported but not used locally
storage/simulation/cluster.go:390:1 func "String" is exported but not used locally
gossip/simulation/network.go:90:1 func "CreateNode" is exported but not used locally
```

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4198)

### CODE REVIEW

File: storage/simulation/cluster.go [hunk 1]
```diff
@@ -34,9 +34,9 @@ import (
 	"github.com/cockroachdb/cockroach/util/stop"
 )
 
-// Cluster maintains a list of all nodes, stores and ranges as well as any
+// simCluster maintains a list of all nodes, stores and ranges as well as any
 // shared resources.
-type Cluster struct {
+type simCluster struct {
```
Comment (bdarnell) [inline storage/simulation/cluster.go]: Why did you add `sim` prefixes in this file instead of simply renaming to lowercase? Also, exports from `package main` are pretty meaningless so I think we should just leave this package alone.

Comment (tamird) [inline storage/simulation/cluster.go]: I had to add the sim prefixes to avoid collisions with variable names, but you're right, I'll revert changes to this package.

**Comment by BramGruneir** [comment]
LGTM
